Automattic/woocommerce-payments
# Fix WooPay trial subscriptions purchases

Fixes #9777

#### Changes proposed in this Pull Request








This PR fixes an error that happens when purchasing only trial subscriptions with WooPay.

The error happens on [this line](https://github.com/woocommerce/woocommerce/blob/1675618a26dc8d2d1026e14d252b9bef80df30c0/plugins/woocommerce/src/StoreApi/Utilities/DraftOrderTrait.php#L64) because WooPay sets the order to a `processing` state during the preflight request and orders with trial subscription has a `0` total price, making the `need_payment()` return `false` because [it expects the order status](https://github.com/woocommerce/woocommerce/blob/1675618a26dc8d2d1026e14d252b9bef80df30c0/plugins/woocommerce/includes/class-wc-order.php#L1787) to be `pending` or `failed`, thus creating a new order making we get the order id mismatch error [here](https://github.com/Automattic/woocommerce-payments/blob/393190472bb2a5c50f9e7addaac6f45018c63841/includes/class-wc-payment-gateway-wcpay.php#L1747).

#### Testing instructions





* Add one or more trial subscriptions to the cart.
* Checkout with WooPay.
* Place order on WooPay.
* The order should be placed successfully.

-------------------


**Post merge**
